File: Data Structures & Algorithms/is-palindrome/submission-4.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isPalindrome(s) {
        let l = 0
        let r = s.length - 1

        while (l < r) {
            if (!this.isAlphaNum(s[l])) {
                l++
                continue
            }
            if (!this.isAlphaNum(s[r])) {
                r--
                continue
            }

            if (s[l].toLowerCase() !== s[r].toLowerCase()) {
                return false
            }
            l++
            r--
        }

        return true
    }

    isAlphaNum(c) {
        const code = c.charCodeAt(0)
        return (
            (code >= 48 && code <= 57) || 
            (code >= 65 && code <= 90) ||
            (code >= 97 && code <= 122)
        )
    }
}

File: Data Structures & Algorithms/palindrome-number/submission-4.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        
        reversedNum = 0
        n = x

        while n:
            reversedNum = reversedNum * 10 + n % 10
            n //= 10

        return x == reversedNum
